modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/util/DDMFormValuesUtil.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.web.internal.util;
+
+import com.liferay.dynamic.data.mapping.form.values.factory.DDMFormValuesFactory;
+import com.liferay.dynamic.data.mapping.kernel.DDMStructure;
+import com.liferay.dynamic.data.mapping.storage.DDMFormValues;
+import com.liferay.dynamic.data.mapping.util.DDMBeanTranslator;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.servlet.DynamicServletRequest;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Tai Pham
+ */
+@Component(service = {})
+public class DDMFormValuesUtil {
```

Comment:
Why is this class necessary? If I'm not mistaken, this is used only from the display context. That class is a proper component, so if you move this logic there there's no need to access DS references statically (which should be a last resort hack).

Reply:
I removed `DDMFormValuesUtil` class and moved the logic to the `DefaultDLEditFileEntryDisplayContext` and updated its constructor. Please help me review it again. Thanks a lot.